### ISSUE TITLE
Check initQpStatefulProperties is being called during evaluation of DiracKernels for compute=false Materials

### ISSUE THREAD

**Issue by WilkAndy** [idaholab/moose#12545]
## Rationale

I suspect it's not getting called, but need to check, and if so, reproduce with a simple input file.

## Description

I wrote:

> 	I believe that initQpStatefulProperties is not being called during evaluation of DiracKernels when the Material is a compute=false Material.  Is this well-known?  If so, how can i overcome it? 

@aeslaughter  wrote

> That function should be called, the logic is here: https://github.com/idaholab/moose/blob/5686e1de83c5553e051fa399b9696e083bd6ebee/framework/src/loops/ComputeMaterialsObjectThread.C#L113
> If you can reproduce the behavior with a simple, MOOSE-only input file we could debug it.

## Impact

Bug fix, if a bug exists

**Comment by WilkAndy** [comment]
Almost definitely at the very start of ```MultiParameterPlasticityStressUpdate::updateState``` i should have
```
  // Size _yf[_qp] appropriately
  // _plastic_strain and _intnl are already sized appropriately because they are stateful
  _yf[_qp].assign(_num_yf, 0);
```
But this does not fix the Dirac problem

**Comment by WilkAndy** [comment]
I think i've got to the bottom of this.  Bad news:

https://github.com/idaholab/moose/blob/next/framework/src/dirackernels/DiracKernel.C#L91-92

I can fix minor things mentioned here and in #12927 and in #12962 by doing some coding to address a corner case of particular interest, but the overall problem remains: stateful materials cannot be used with DiracKernels.

**Comment by WilkAndy** [comment]
Since DiracKernels do not currently support stateful Materials, i'm closing this issue.  Stateful+Dirac would be a great addition to the MOOSE framework.
